### PR TITLE
Sharing scope

### PR DESCRIPTION
**Pull request by rpiazza**

### CODE REVIEW

File: stix_edh/bindings/cyber_profile.py [hunk 1]
```diff
@@ -884,4 +891,4 @@ def parseString(inString):
     "PolicyType",
     "PublicReleaseType",
     "ResourceDispositionType"
-]
+    ]
```
Comment (clenk) [inline stix_edh/bindings/cyber_profile.py]: Don't think this indentation is necessary.

File: stix_edh/bindings/cyber_profile.py [hunk 2]
```diff
@@ -277,7 +279,7 @@ def buildChildren(self, child_, node, nodeName_, fromsubclass_=False):
             self.gds_validate_string(child_.text, node, 'compilationReason')
             obj_ = edh_common.NMTOKENS.factory()
             obj_.build(child_)
-            self.compilationReason = obj_
+            self.set_compilationReason(obj_)
```
Comment (clenk) [inline stix_edh/bindings/cyber_profile.py]: Is this related to sharing scope? This change makes me nervous; not sure what it impacts.

File: stix_edh/bindings/cyber_profile.py [hunk 3]
```diff
@@ -442,7 +444,6 @@ def exportChildren(self, lwrite, level, nsmap, namespace_=XML_NS, name_='Declass
             showIndent(lwrite, level, pretty_print)
             lwrite('<%s:declassEvent>%s</%s:declassEvent>%s' % (nsmap[namespace_], quote_xml(self.declassEvent), nsmap[namespace_], eol_))
     def build(self, node):
-        self.__sourcenode__ = node
```
Comment (clenk) [inline stix_edh/bindings/cyber_profile.py]: Is this related to sharing scope? All the other `build()` functions include this line so I'm nervous about removing it here.

File: stix_edh/bindings/cyber_profile.py [hunk 4]
```diff
@@ -817,9 +823,10 @@ def buildAttributes(self, node, attrs, already_processed):
         super(FurtherSharingType, self).buildAttributes(node, attrs, already_processed)
     def buildChildren(self, child_, node, nodeName_, fromsubclass_=False):
         if nodeName_ == 'sharingScope':
-            sharingScope_ = child_.text
-            sharingScope_ = self.gds_validate_string(sharingScope_, node, 'sharingScope')
-            self.sharingScope = sharingScope_
+            self.gds_validate_string(child_.text, node, 'sharingScope')
```
Comment (clenk) [inline stix_edh/bindings/cyber_profile.py:827]: The previous version of the code saved the result to a variable and used it. Is it okay that we aren't doing that?

File: stix_edh/bindings/cyber_profile.py (hunk 1 above)
Comment (rpiazza) [reply stix_edh/bindings/cyber_profile.py]: I didn't give you all of the details about creating this file.  I actually did a diff and brought over the original code where it was different.  So some of these differences are because of that.  I will remove this difference.

File: stix_edh/bindings/cyber_profile.py (hunk 3 above)
Comment (rpiazza) [reply stix_edh/bindings/cyber_profile.py]: Same - didn't bring it over

File: stix_edh/bindings/cyber_profile.py (hunk 4 above)
Comment (rpiazza) [reply stix_edh/bindings/cyber_profile.py:827]: Actually, the generated code for this didn't work.  I'm not sure why - so I "stole" this code from AccessPrivilegeType, which has a similar property to sharingScope (see Line 724)

Comment (rpiazza) [reply stix_edh/bindings/cyber_profile.py:827]: sharingScope_ = child_.text
            sharingScope_ = self.gds_validate_string(sharingScope_, node, 'sharingScope')
            self.sharingScope.append(sharingScope_)

**Review by clenk — APPROVED** [review]
LGTM
